sw/device/lib/testing/check.h

```diff
@@ -53,33 +53,41 @@
  * Prints differences between `actual_` and `expected_` before assigning the
  * comparison `result_` value.
  *
- * @param[out] result_ Result of the comparison.
  * @param actual_ Buffer containing actual values.
  * @param expected_ Buffer containing expected values.
  * @param num_items_ Number of items to compare.
+ * @param ... Arguments to a LOG_* macro, which are evaluated if the check.
  */
-#define CHECK_BUFFER(result_, actual_, expected_, num_items_)                 \
-  do {                                                                        \
-    /* `sizeof(actual_[0])` is used to determine the size of each item in the \
-     * buffer. */                                                             \
-    if (memcmp(actual_, expected_, num_items_ * sizeof(actual_[0])) != 0) {   \
-      for (size_t i = 0; i < num_items_; ++i) {                               \
-        LOG_ERROR("[%d] actual = 0x%x; expected = 0x%x", i, actual_[i],       \
-                  expected_[i]);                                              \
-      }                                                                       \
-      result_ = false;                                                        \
-    } else {                                                                  \
-      result_ = true;                                                         \
-    }                                                                         \
+#define CHECK_BUFFER(actual_, expected_, num_items_, ...)                      \
+  do {                                                                         \
+    /* `sizeof(actual_[0])` is used to determine the size of each item in the  \
+     * buffer. */                                                              \
+    if (memcmp(actual_, expected_, num_items_ * sizeof(actual_[0])) != 0) {    \
+      for (size_t i = 0; i < num_items_; ++i) {                                \
+        LOG_INFO("[%d] actual = 0x%x; expected = 0x%x", i, actual_[i],         \
```

Comment:
nit: shouldn't this remain `LOG_ERROR` given that it's printed only if the comparison fails? Alternatively you can use LOG_ERROR only for the elements that are different. Would make spotting differences much easier.

Reply:
DV sim exits on first encountered error log (this is a command line setting). So the rest of the buffer mismatch will not be printed.

Reply:
Yes, that's why I changed to `LOG_INFO`. Happy to change it back to `LOG_ERROR`, that will require DV sim to be re-run with a different cmd line setting to print the actual buffer differences if a test fails here (assuming I put the for loop after the initial LOG_ERROR stating there was a mismatch).

Reply:
SG, I was thinking about FPGA.

Reply:
How about `LOG_INFO("Error: mismatch...` should work as well right? Still an info log, but slightly better....

Reply:
Spoke with @alphan. He is ok with leaving as is. Marking as resolved.